### PR TITLE
Add calcium tablets

### PR DESCRIPTION
**Pull request by BrettDong**

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Do we really need a separate vitamin item for each deficiency?

I'd rather get rid of overdose mechanic and just make multivitamin the sole artificial source of vitamins.

**Comment by kevingranade** [comment]
On Jan 2, 2017 10:23 AM, "Coolthulhu" <notifications@github.com> wrote:

Do we really need a separate vitamin item for each deficiency?

The only question in my mind is whether they're common irl.  For calcium
the answer is yes, for specific vitamins it's mostly no.

**Comment by WizardOfOoo** [comment]
Over-the-counter antacids like Tums and Rolaids, contain calcium carbonate which is an ok source of calcium.

**Comment by WizardOfOoo** [comment]
I can talk about vitamin availability IRL.

Now this is assuming that we are excluding health food stores and pharmacies.  Each has a wide selection of vitamins and minerals, usually in vast quantity.

- Calcium: Like KG said, calcium is very common.  it's in over the counter Antacids like tums or rollaids.

- Iron: Iron poisoning can be an issue, so over the counter iron supplements usually are not very potent or common.  Iron is added to a ton of foods, the need for a supplement is not very high in healthy people.

- A: Vitamin A is going to be the hardest to find in tablet form.  About the only non-food source of vitamin A in the cataclysm would be Fish oil pills or some beauty products (yuck!).  Our diet is so Vitamin A rich that it's uncommon to be deficient in healthy people.

- B complex:  Most energy drinks contain at 1x-3x the daily recommended amount of B6 and B12 vitamins per serving (usually two or three servings in a can)

- As far as C, You can find "immune system boosters" with mega doses of vitamin C in a lot of places.
image A lot of sugary drinks contain vitamin C, it makes them taste tart.

**Comment by Cyrano7** [comment]
Maybe in the description we could call them antacid tablets instead as those are far more common then strait calcium tablets and say that there high calcium content could act as a supplement. I'd leave the actual name calcium tablets though.

**Comment by Reclusive-reptile** [comment]
Brewers yeast is often used to supplement vitamin b complex iirc.

**Comment by mugling** [comment]
> Do we really need a separate vitamin item for each deficiency?
> 
> I'd rather get rid of overdose mechanic and just make multivitamin the sole artificial source of vitamins.

We should get rid of the overdose mechanic as well as all artificial sources (including multivitamins). There's no value in having a system of deficiencies as a challenge for the player if we immediately provide an easy solution to it.

Either vitamin tablets are common IRL so there's no need for a deficiency system (which brings us back to food being a broken game mechanic) or we can presume they've all been scavenged/decayed or otherwise hand-waved away.

Our current implementation satisfies nobody - a vitamin system with no enabled penalties, a mod to disable it, and no real gameplay value if the game contains multivitamins. IIRC we also already have several identical types of multivitamin?

**Comment by WizardOfOoo** [comment]
> Our current implementation satisfies nobody 

I like it, I also I understand that it is a work in progress.  Like everything else.

**Comment by kevingranade** [comment]
> We should get rid of the overdose mechanic as well as all artificial sources (including multivitamins). There's no value in having a system of deficiencies as a challenge for the player if we immediately provide an easy solution to it.

It would be reasonable to turn off the entire vitamin system as "not worthwhile to track since it's too easy too keep up with" (I'd disagree with the argument but it would at least be consistent), but it is absurd to eliminate a major solution to a problem just because it's inconvinient for your implementation.  Effective suppliments for various vitamins and minerals exist, you either need to either come up with a solution that is effective dispite this, or give up on the vitamin system, pretending they don't exist is not an option.

**Comment by mugling** [comment]
> but it is absurd to eliminate a major solution to a problem just because it's inconvinient for your implementation.

Why is it absurd? We made such a gameplay decision for ranged weapons. Rule of fun should apply.

**Comment by mugling** [comment]
> but it is absurd to eliminate a major solution to a problem just because it's inconvinient for your implementation.

Why is it absurd? For example we made such a gameplay decision for ranged weapons. Rule of fun should apply.

**Comment by mugling** [comment]
> but it is absurd to eliminate a major solution to a problem just because it's inconvinient for your implementation.

Why is it absurd? For example we made such a gameplay decision for ranged weapons. Rule of fun should apply.

**Comment by kevingranade** [comment]
You have to be more specific, the fact that the range of guns is scaled down to fit in the visible area around the player, or something else?
You aren't invoking the "rule of fun" you're invoking the "rule of I want this system to hurt the player so I'm taking away some of their options".

**Comment by Coolthulhu** [comment]
> it is absurd to eliminate a major solution to a problem just because it's inconvinient for your implementation

It's inconvenient for the game to have features that amount only to keypresses.
For example, having to track 10 nutrition meters, each of which is trivial to fill, is about as fun as being reminded to clip nails and empty the bladder and taking penalties for not doing so.

We don't have an auto-eat feature. With one, it could be manageable. Pointless, but manageable.
Without one but with trivially hoardable pills, the only result would be having to manually consume the pills once in a while. Which would add nothing to the game - nothing good at the very least.

There are two sides of rule of fun: one is to add things that are fun, the other is not to add things that are anti-fun.

**Comment by WizardOfOoo** [comment]
I find vitamins to be fun.

**Comment by BrettDong** [comment]
There is a med product in our country that is specially designed to supplement calcium, mainly targeting the elderly people. It can be easily found in pharmacies and even normal stores in the cities, very popular and sells very well.

cal

Don't know much about medicines in America, but I assume there is also some kind of calcium supplementary tablets that are sold widely?

**Comment by Rivet-the-Zombie** [comment]
>Don't know much about medicines in America, but I assume there is also some kind of calcium supplementary tablets that are sold widely?

Yes, lots of different ones.

**Comment by kevingranade** [comment]
On Jan 3, 2017 1:22 AM, "Coolthulhu" <notifications@github.com> wrote:

it is absurd to eliminate a major solution to a problem just because it's
inconvinient for your implementation

It's inconvenient for the game to have features that amount only to
keypresses.

You trimmed the part where I said
> It would be reasonable to turn off the entire vitamin system as "not
worthwhile to track since it's too easy too keep up with"

The point of contention is the suggestion to leave the vitamin system in
but remove multivitamins.  That would just be stupid, considering
multivitamins are a well-known solution to this very problem, and people
would be justifiably annoyed at that solution being removed to make the
game arbitrarily more difficult.

**Comment by BrettDong** [comment]
Keeping the vitamins sufficiently fed with the presence of multivitamins is already a challenging task for players. I set up a vote surveying the public idea in a Chinese community yesterday, one day past, now the result suggests that the idea of removing those nutrition supplementary items is strongly opposed by the players (0 agree, 2 neutral, 21 reject).

**Comment by kevingranade** [comment]
While there are valid questions about the vitamin system as a whole, there was nothing raised about this particular change, merging.
If you have specific issues or proposed solutions about the vitamin system in general, please open a dedicated issue or PR.

File: data/json/items/comestibles/med.json [hunk 1]
```diff
@@ -1208,6 +1208,28 @@
         }
     },
     {
+        "id" : "calcium_tablet",
+        "type" : "COMESTIBLE",
+        "comestible_type" : "MED",
+        "name" : "calcium tablet",
+        "description" : "White calcium tablets.  Widely used by elderly people with osteoporosis as a method to supplement calcium before apocalypse.",
```
Comment (Cyrano7) [inline data/json/items/comestibles/med.json]: before the* apocalypse

**Pull request merged**